radio/src/os/task.h

```diff
@@ -43,14 +43,74 @@ bool mutex_lock(mutex_handle_t* h);
 void mutex_unlock(mutex_handle_t* h);
 bool mutex_trylock(mutex_handle_t* h);
 
+
+// this helper class is only to be used on the stack and never to be shared outside the scope where it was created
+/* example usage
+ *
+ * void doWork()
+ * {
+ *   MutexLock lock = MutexLock::MakeInstance(mutex);
+ *   while(!queue.isEmpty)
+ *   {
+ *     lock.unlock();
+ *     auto item = queue.pop();
+ *     <do something with item>
+ *     lock.lock();
+ *   }
+ * } // auto mutex unlock, because lock is destructed here
+ *
+ * int sendMessage()
+ * {
+ *   MutexLock lock = MutexLock::MakeInstance(busMutex);
```

Comment:
Here you could start a block with the MutexLock inside. No need for `lock` and `unlock` functions

Reply:
That is not correct. You have to protect the queue.empty() check against other queue accesses, like a queue push from another thread.